lightning/src/ln/msgs.rs

```diff
@@ -1515,6 +1580,165 @@ impl_writeable_len_match!(NodeAnnouncement, {
+		if encoding_type != EncodingType::Uncompressed as u8 {
+			return Err(DecodeError::InvalidValue);
+		}
+		encoding_len -= 1;
+		let mut read_len = 0;
+		let mut short_channel_ids = vec![];
+		loop {
+			if read_len >= encoding_len { break; }
+			short_channel_ids.push(Readable::read(r)?);
+			read_len += 8;
+		}
+
+		Ok(QueryShortChannelIds {
+			chain_hash,
+			short_channel_ids,
+		})
+	}
+}
+
+impl Writeable for QueryShortChannelIds {
+	fn write<W: Writer>(&self, w: &mut W) -> Result<(), ::std::io::Error> {
+		let encoding_len: u16 = 1 + self.short_channel_ids.len() as u16 * 8;
+		w.size_hint(32 + 2 + encoding_len as usize);
+		self.chain_hash.write(w)?;
+
+		encoding_len.write(w)?;
+		(EncodingType::Uncompressed as u8).write(w)?;
+		for scid in self.short_channel_ids.iter() {
+			scid.write(w)?;
+		}	
+
+		Ok(())
+	}
+}
+
+impl Readable for ReplyShortChannelIdsEnd {
+	fn read<R: Read>(r: &mut R) -> Result<Self, DecodeError> {
+		let chain_hash: BlockHash = Readable::read(r)?;
+		let full_information: bool = Readable::read(r)?;
+		Ok(ReplyShortChannelIdsEnd {
+			chain_hash,
+			full_information,
+		})
+	}
+}
+
+impl Writeable for ReplyShortChannelIdsEnd {
+	fn write<W: Writer>(&self, w: &mut W) -> Result<(), ::std::io::Error> {
+		w.size_hint(32 + 1);
+		self.chain_hash.write(w)?;
+		self.full_information.write(w)?;
+		Ok(())
+	}
+}
+
+impl Readable for QueryChannelRange {
+	fn read<R: Read>(r: &mut R) -> Result<Self, DecodeError> {
+		let chain_hash: BlockHash = Readable::read(r)?;
+		let first_blocknum: u32 = Readable::read(r)?;
+		let number_of_blocks: u32 = Readable::read(r)?;
+		Ok(QueryChannelRange {
+			chain_hash,
+			first_blocknum,
+			number_of_blocks
+		})
+	}
+}
+
+impl Writeable for QueryChannelRange {
+	fn write<W: Writer>(&self, w: &mut W) -> Result<(), ::std::io::Error> {
+		w.size_hint(32 + 4 + 4);
+		self.chain_hash.write(w)?;
+		self.first_blocknum.write(w)?;
+		self.number_of_blocks.write(w)?;
+		Ok(())
+	}
+}
+
+impl Readable for ReplyChannelRange {
+	fn read<R: Read>(r: &mut R) -> Result<Self, DecodeError> {
+		let chain_hash: BlockHash = Readable::read(r)?;
+		let first_blocknum: u32 = Readable::read(r)?;
+		let number_of_blocks: u32 = Readable::read(r)?;
+		let full_information: bool = Readable::read(r)?;
+
+		let mut encoding_len: u16 = Readable::read(r)?;
+		let encoding_type: u8 = Readable::read(r)?;
+		if encoding_type != EncodingType::Uncompressed as u8 {
+			return Err(DecodeError::InvalidValue);
+		}
+		encoding_len -= 1;
+		let mut read_len = 0;
+		let mut short_channel_ids = vec![];
+		loop {
+			if read_len >= encoding_len { break; }
+			short_channel_ids.push(Readable::read(r)?);
+			read_len += 8;
+		}
+
+		Ok(ReplyChannelRange {
+			chain_hash,
+			first_blocknum,
+			number_of_blocks,
+			full_information,
+			short_channel_ids
+		})
+	}
+}
+
+impl Writeable for ReplyChannelRange {
+	fn write<W: Writer>(&self, w: &mut W) -> Result<(), ::std::io::Error> {
+		w.size_hint(32 + 4 + 4);
```

Comment:
Why `size_hint` a first time ?

Reply:
Bah, oversight on my part. It's been removed.